STUDIO/org.openl.rules.project.openapi/src/org/openl/rules/project/openapi/OpenApiGenerator.java

```diff
@@ -0,0 +1,322 @@
+package org.openl.rules.project.openapi;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openl.CompiledOpenClass;
+import org.openl.classloader.OpenLClassLoader;
+import org.openl.rules.lang.xls.binding.XlsModuleOpenClass;
+import org.openl.rules.project.IRulesDeploySerializer;
+import org.openl.rules.project.instantiation.RulesInstantiationException;
+import org.openl.rules.project.instantiation.RulesInstantiationStrategy;
+import org.openl.rules.project.instantiation.RuntimeContextInstantiationStrategyEnhancer;
+import org.openl.rules.project.instantiation.variation.VariationInstantiationStrategyEnhancer;
+import org.openl.rules.project.model.ProjectDescriptor;
+import org.openl.rules.project.model.RulesDeploy;
+import org.openl.rules.project.resolving.ProjectResource;
+import org.openl.rules.project.resolving.ProjectResourceLoader;
+import org.openl.rules.project.xml.XmlRulesDeploySerializer;
+import org.openl.rules.ruleservice.core.RuleServiceInstantiationFactoryHelper;
+import org.openl.rules.ruleservice.core.interceptors.DynamicInterfaceAnnotationEnhancerHelper;
+import org.openl.rules.ruleservice.publish.jaxrs.JAXRSOpenLServiceEnhancerHelper;
+import org.openl.rules.ruleservice.publish.jaxrs.swagger.OpenApiObjectMapperHack;
+import org.openl.rules.ruleservice.publish.jaxrs.swagger.OpenApiRulesCacheWorkaround;
+import org.openl.rules.ruleservice.publish.jaxrs.swagger.SchemaJacksonObjectMapperFactoryBean;
+import org.openl.rules.ruleservice.publish.jaxrs.swagger.jackson.OpenApiObjectMapperConfigurationHelper;
+import org.openl.rules.ruleservice.publish.jaxrs.swagger.jackson.OpenApiObjectMapperFactory;
+import org.openl.types.IOpenClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.swagger.v3.jaxrs2.Reader;
+import io.swagger.v3.oas.models.OpenAPI;
+
+public class OpenApiGenerator {

```

Comment:
JavaDocs about what this class does. Input data (e.g. Excel or OpenL Projects), output data (String, Stream, File...). Commonly, All public and protected methods should have a short description.